gensim/corpora/textcorpus.py

```diff
@@ -575,62 +610,89 @@ def getstream(self):
+            source (str): path to a directory where all files can be opened by the
+                `LineSentence` class. Each file will be read up to `limit` lines
+                (or all lines if `limit` is None, the default). The files in the
+                directory should be either text files, .bz2 files, or .gz files.
+            sort_filenames (bool): whether or not to sort the filenames read from
+                the directory. If True (default), all filenames will be read into
+                memory on each call to `iter_filepaths` (called by `getstream`).
+                If False, the files are read in the order they are encountered during
+                the top-down directory walk.
 
         Example::
 
             sentences = LineSentencePath(os.getcwd() + '\\corpus\\')
 
-        The files in the directory should be either text files, .bz2 files, or .gz files.
-
         """
-        self.source = source
+        if os.path.isfile(source):
+            raise ValueError(
+                "source '%s' is file; use `corpora.textcorpus.LineSentence` instead" % source)
+
         self.max_sentence_length = max_sentence_length
         self.limit = limit
+        self.sort_filenames = sort_filenames
+        kwargs['lines_are_documents'] = True
+        kwargs['tokenizer'] = kwargs.get('tokenizer', unicode_and_tokenize)
+        kwargs['character_filters'] = kwargs.get('character_filters', [])
+        kwargs['token_filters'] = kwargs.get('token_filters', [])
+        kwargs['processes'] = kwargs.get('processes', 1)
+        super(PathLineSentences, self).__init__(source, **kwargs)
 
-        if os.path.isfile(self.source):
-            logging.warning('single file read, better to use models.word2vec.LineSentence')
-            self.input_files = [self.source]  # force code compatibility with list of files
-        elif os.path.isdir(self.source):
-            self.source = os.path.join(self.source, '')  # ensures os-specific slash at end of path
-            logging.debug('reading directory ' + self.source)
-            self.input_files = os.listdir(self.source)
-            self.input_files = [self.source + file for file in self.input_files]  # make full paths
-            self.input_files.sort()  # makes sure it happens in filename order
-        else:  # not a file or a directory, then we can't do anything with it
-            raise ValueError('input is neither a file nor a path')
+    def getstream(self):
+        """Yield documents from the underlying plain text collection (of one or more files).
+        Each item yielded from this method will be considered a document by subsequent
+        preprocessing methods.
+        """
+        paths = self.iter_filepaths()
+        if self.sort_filenames:
+            logger.debug("sorting filepaths")
+            paths = list(paths)
+            paths.sort(key=lambda path: os.path.basename(path))
+            logger.debug("found {} files: {}".format(len(paths), paths))
```

Comment:
The intention here was to get the auto-formatting for the list of paths, as opposed to having to do my own '[' + ', '.join(paths) + ']', which seemed much messier. Should I still change it do this instead?